src/sfdc/base/main/default/classes/B2CCustomerAddressBook.cls

```diff
@@ -241,12 +253,16 @@ public with sharing class B2CCustomerAddressBook {
         contactQuery.selectField('AccountId');
         contactQuery.selectField('B2C_Customer_No__c');
         contactQuery.selectField('B2C_CustomerList__r.Name');
+        contactQuery.selectField('B2C_CustomerList__r.Is_Active__c');
         contactQuery.selectField('B2C_CustomerList__r.Base_API_Url__c');
         contactQuery.selectField('B2C_CustomerList__r.API_Url__c');
         contactQuery.selectField('B2C_CustomerList__r.B2C_Client_ID__c');
         contactQuery.selectField('B2C_CustomerList__r.OCAPI_Version_No__c');
+        contactQuery.selectField('B2C_CustomerList__r.Enable_Address_Publishing__c');
+        contactQuery.selectField('B2C_CustomerList__r.Enable_Address_Retrieval__c');
         contactQuery.setLimit(1);
         contactQuery.addConditionEq('B2C_Customer_Id__c', customerId);
+        contactQuery.addConditionEq('B2C_CustomerList__r.Is_Active__c', true);
```

Comment:
This is causing the wrong contact to be retrieved because the query becomes `WHERE (B2C_Customer_Id__c = '0031F00000myBC1QAM') OR (B2C_CustomerList__r.Is_Active__c = true) OR (Id = '0031F00000myBC1QAM')...`